### DISCUSSION TITLE
Moving agents between simulations

### DISCUSSION THREAD

**Discussion by leenr01** [gama-platform/gama#67] [Q&A] [answered]
I want to simulate multiple cities each being one separate simulation. I created the simulations using a GUI experiment.
Is there a way I can move agents between these simulations while preserving the agent properties?
I believe this -if possible- should be done in the experiment part and not the species part of the code.

**Comment by ptaillandier** [comment, marked as answer]
Hi,

A simple solution is to create the same agent in the new simulation. Here
is an example.

```

model movebetweensimulation

global {
	int grid_size <- 10;
	
	init {
		if int(self) = 0 {
			create inhabitant with: (location:location);
		}
	}
}


species inhabitant skills: [moving]{
	rgb color <- rnd_color(255);
	int simulation_id <- 0;
	reflex move {
		do move heading: 0.0;
		if (location.x = world.shape.width) {
			simulation_id <- (simulation_id +1) mod length(experiment.simulations);
			ask experiment.simulations[simulation_id]{
				create inhabitant with:(location:{0.0, location.y}, color:color, simulation_id:myself.simulation_id); 
			}
			do die;
		}
	}
	
	aspect default {
		draw circle(2) color: #red;
	}
}

grid cell width: grid_size height: grid_size;

experiment movebetweensimulation type: gui autorun: true {
	float minimum_cycle_duration <- 0.02;
	action _init_ {
		create simulation with: 	(grid_size: 10);
		create simulation with: 	(grid_size: 5);
	}
	
	output {
		display map {
			grid cell border: #black;
			species inhabitant;
		}
	}
}

```

Cheers,

Patrick

**Reply by leenr01** [reply]
Thank you! That was helpful.

**Reply by leenr01** [reply]
Hello,

When I tried your method, the function "count" does not work anymore as it is trying to count dead agents as well.
I am getting the error: Cannot evaluate "what I am trying to count" as the target agent is dead.

I tried counting only alive agents but it did not work as it has to go over all agents first so the same problem occurs.
Do you have any idea on how to erase dead agents completely?

**Comment by benoitgaudou** [comment]
Hi
There is also a model in the model library related to network (MQTT), in
which an agent is sent from one simulation to another simulation

Cheers

Benoit


Le lun. 13 mars 2023 à 02:12, ptaillandier ***@***.***> a
écrit :

> Hi,
>
> A simple solution is to create the same agent in the new simulation. Here
> is an example.
>
> Cheers,
>
> Patrick
>

**Reply by leenr01** [reply]
Thank you dear Benoit.
